skbio/stats/composition.py

```diff
@@ -2058,3 +2060,489 @@ def dirmult_ttest(
         "Reject null hypothesis",
     ]
     return res[col_order]
+
+
+def _type_cast_to_float(df):
+    """Attempt to cast all of the values in dataframe to float.
+
+    This will try to type cast all of the series within the
+    dataframe into floats.  If a column cannot be type casted,
+    it will be kept as is.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+
+    Returns
+    -------
+    pd.DataFrame
+    """
+    # Implementation based on https://github.com/mortonjt/differential/blob/65752567ef4cf303471405b0a9be503eb10a0bbb/differential/util.py#L4
+    # TODO: Will need to improve this, as this is a very hacky solution.
+    for c in df.columns:
+        s = df[c]
+        try:
+            df[c] = s.astype(np.float64)
+        except Exception:
+            continue
+    return df
+
+
+def _lme_call(
+    formula,
+    metadata,
+    table,
+    groups,
+    reml=True,
+    method=None,
+    re_formula=None,
+    vc_formula=None,
+    fit_kwargs={},
+    **kwargs,
+):
+    # TODO: add docs when this implementation is approved
+    # _covariate_list is essentially the list of covariates
+
+    FEATUREID = "FeatureID"
+    LOG2FC = "Log2(FC)"
+    CI25 = "CI(2.5)"
+    CI975 = "CI(97.5)"
+    PVALUE = "pvalue"
+    COVARIATE = "Covariate"
+
+    submodels = []
+    metadata = _type_cast_to_float(metadata.copy())
+
+    merged_data = pd.merge(table, metadata, left_index=True, right_index=True)
+    if len(merged_data) == 0:
+        raise ValueError(
+            (
+                "No more samples left. Check to make sure that "
+                "the sample names between `metadata` and `data` "
+                "are consistent."
+            )
+        )
+
+    design_matrix = dmatrix(formula, metadata, return_type="dataframe")
+
+    # Obtaining the list of covariates by selecting the relevant columns
+    _covariate_list = design_matrix.columns.tolist()
+
+    # Removing intercept since it is not a covariate, and is included by default
+    _covariate_list.remove("Intercept")
+
+    output = []
+    for response_var in table.columns:
+        # mixed effects code is obtained here:
+        # http://stackoverflow.com/a/22439820/1167475
+        stats_formula = "%s ~ %s" % (response_var, formula)
+        model = MixedLM.from_formula(
+            formula=stats_formula,
+            data=merged_data,
+            groups=groups,
+            re_formula=re_formula,
+            vc_formula=vc_formula,
+            **kwargs,
+        )
+
+        results = model.fit(reml=reml, method=method, **fit_kwargs)
```

Comment:
I don't believe that we need to explicitly expose `reml` in the args. If we just remove this in all of the function signatures, users should still be able to pass `reml` in to the `fit` method via the `fit_kwargs`.

I suggest deleting `reml` from all of the function signatures, but keep the tests the same to confirm.

Reply:
I have made the necessary changes by removing `reml` from all function signatures as recommended. The functionality remains intact, and the updated function call is 
```python
fit_kwargs = {"reml": False}
res_ml = dirmult_lme(formula=formula, data=self.table, metadata=self.metadata, groups='Covar1', seed=0, p_adjust=None, fit_kwargs=fit_kwargs)
```